fullstack/freertos-wolfip-wolfmqtt/README.md

```diff
@@ -0,0 +1,125 @@
+# FreeRTOS + wolfIP + wolfMQTT TLS Example
+
+This example demonstrates a full-stack embedded MQTT client using FreeRTOS, wolfIP, and wolfMQTT with TLS security. It shows how to establish a secure MQTT connection from an embedded device to a broker using wolfSSL for TLS.
+
+## Software Bill of Materials (SBOM)
+
+| Component | Version | License | Description |
+|-----------|---------|---------|-------------|
+| FreeRTOS | 10.5.1 | MIT | Real-time operating system for embedded devices |
+| wolfSSL | 5.6.3 | GPLv2 | Embedded SSL/TLS library |
+| wolfMQTT | 1.16.0 | GPLv2 | Embedded MQTT client library |
+| wolfIP | 1.0.0 | GPLv2 | Embedded TCP/IP stack |
+| Mosquitto | 2.0.15 | EPL/EDL | MQTT broker for testing |
+
+## Prerequisites
+
+- Linux development environment
+- CMake 3.13 or later
+- GCC compiler
+- Mosquitto MQTT broker
+- Root/sudo privileges (for TAP interface setup)
+
+## Setup
+
+1. Clone the required repositories:
+   ```
+   ./setup.sh
+   ```
+
+   This script will:
+   - Clone FreeRTOS kernel
+   - Clone FreeRTOS repository
+   - Create necessary directories
+   - Configure the build environment
+
+2. Configure the TAP interface on your host system:
+   ```
+   sudo ip tuntap add dev tap0 mode tap
+   sudo ip addr add 10.10.0.1/24 dev tap0
+   sudo ip link set dev tap0 up
+   ```
+
+3. Configure Mosquitto broker with TLS:
```

Comment:
It would be less invasive to launch an instance of mosquitto configured to listen on a different port (eg. `118883`). That would remove the need to edit a system file and mess with the mosquitto service.